File: root_quadratic.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    double a, b, c;
    cout << "Enter the coefficient of x^2: ";
    cin >> a;
    cout << "Enter the coefficient of x: ";
    cin >> b;
    cout << "Enter the constant: ";
    cin >> c;
    double discriminant = b*b - 4*a*c;
    if (discriminant < 0) {
        cout << "The equation has no real roots." << endl;
    } else if (discriminant == 0) {
        double root = -b / (2 * a);
        cout << "The equation has one real root: " << root << endl;
    } else {
        double root1 = (-b + sqrt(discriminant)) / (2 * a);
        double root2 = (-b - sqrt(discriminant)) / (2 * a);
        cout << "The equation has two real roots: " << root1 << " and " << root2 << endl;
    }
    return 0;
}

File: root_quadratic.js
function quadraticRoots(a, b, c) {
  let root1, root2;
  let determinant = b * b - 4 * a * c;
  if (determinant > 0) {
    root1 = (-b + Math.sqrt(determinant)) / (2 * a);
    root2 = (-b - Math.sqrt(determinant)) / (2 * a);
    console.log("The roots are " + root1 + " and " + root2);
  } else if (determinant == 0) {
    root1 = root2 = -b / (2 * a);
    console.log("The root is " + root1);
  } else {
    console.log("The equation has no real roots");
  }
}

quadraticRoots(1, -5, 6);
